torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -2602,30 +2604,53 @@ def is_strict_for(v: VariableTracker):
             bwd_node,
             *([arg.as_proxy() for arg in filtered_args] + list(fwd_freevars.keys())),
         )
-        example_value = pytree.tree_map_only(
-            torch.fx.Proxy,
-            lambda a: a.node.meta["example_value"],
-            fwd_out.as_proxy(),
-        )
+        kwargs = {
+            "args_tensor_mask": args_tensor_mask,
+            "non_differentiable_idx": non_differentiable_idx,
+        }
 
         # Store the invocation as a call
         from torch._functorch.autograd_function import autograd_function_apply
 
+        # We use speculate_subgraph to get the fwd graph, but it's alway under no grad mode like what eager mode does.
```

Comment:
Btw I reverted something that this PR depends on when you were out so that we can have consistency with the release branch and the main branch (https://github.com/pytorch/pytorch/pull/137891). So we should check that we're including it in this change